third_party/amd/lib/TritonAMDGPUTransforms/StreamPipeline.cpp

```diff
@@ -562,7 +562,7 @@ void LoopPipeliner::emitEpilogue(DenseMap<Value, Value> &newResults) {
     epilogueMap.map(load.value()->getResult(0),
                     pplForOp.getResult(bufferIdx + load.index()));
   // Map IV to original upper bound (ie. last iteration)
-  epilogueMap.map(forOp.getInductionVar(), forOp.getUpperBound());
+  epilogueMap.map(forOp.getInductionVar(), pplForOp.getUpperBound());
```

Comment:
I don't know the details of this pass but it seems off to assume that the induction variable is equal to the upper bound for the last iteration.
I was expecting it to be something like: 
```
lb + ((ub - 1 - lb)/step)*step
```
Although since it has been working I'm guessing I'm missing something

Reply:
Hi @ThomasRaoux , you are right. This only works if the `step` is 1. I will amend this and add a further test case

Reply:
thanks. For step 1 wouldn't the last iteration have `iv = ub -1`?

Reply:
Correct, and this is true for `pplForOp.getUpperBound()` (we are using the pipelined loop upper bound, and not the original loop). Which brings me to another question: should we special case for step==1, or can we relay on constant propagation to do the job?

Reply:
ah I get it now, it is because the pipelined loop has one less iteration? Ok I get it now. Yeah would be easier to read by doing the math explicitly from the original loop bound

Reply:
> should we special case for step==1, or can we relay on constant propagation to do the job?

We should rely on constant propagation

Reply:
Thanks for the comments @ThomasRaoux ! I fixed the code and changed the test. PTAL

Reply:
Thanks, that looks good to me!